### ISSUE TITLE
implement correctly static analysis rules suppression

### ISSUE THREAD

**Issue by iXamDev** [lejard-h/chopper#506]
## Steps to Reproduce

1. Create any chopper service  
2. Adjust `analysis_options.yaml` and enable non-default rules, for example, 'omit_local_variable_types'
3. Proceed code generation
4. Execute `flutter analyze --fatal-warnings` on the code sample 

**Expected results:**
No warnings or errors are shown for generated `.chopper.dart` files

**Actual results:**

Linter warnings are shown for `.chopper.dart` files:

  info • Unnecessary type annotation on a local variable • sources/core/patient_core/lib/src/api/services/dummy_service.chopper.dart:21:5 • omit_local_variable_types

<img width="629" alt="image" src="https://github.com/lejard-h/chopper/assets/46491005/e474ebaf-316f-494b-85e9-a8080a11ad48">

<details>
<summary>Code sample</summary>

`dummy_service.dart`

```dart
import 'package:chopper/chopper.dart';

part 'dummy_service.chopper.dart';

@ChopperApi(baseUrl: 'dummy')
abstract class DummyService extends ChopperService {
  static DummyService create([ChopperClient? client]) => _$DummyService(client);

  @Get(path: 'sample')
  Future<Response<dynamic>> get();
}
```

`analysis_options.yaml`

```
include: package:flutter_lints/flutter.yaml

linter:
  rules:
    omit_local_variable_types: true
```

</details>

<details>
  <summary>Logs</summary>

```
   info • Unnecessary type annotation on a local variable • sources/core/platform_core/lib/src/api/services/dummy_service.chopper.dart:21:5 • omit_local_variable_types
   info • Unnecessary type annotation on a local variable • sources/core/platform_core/lib/src/api/services/dummy_service.chopper.dart:22:5 • omit_local_variable_types
```

</details>

## Solution

It would help if you disabled the linter for the generated files entirely instead of ignoring specific rules.

So please replace the comment in generated files `// ignore_for_file: always_put_control_body_on_new_line, always_specify_types, prefer_const_declarations, unnecessary_string_interpolations, unnecessary_brace_in_string_interps
` with `// ignore_for_file: type=lint`. 

This approach is used by other projects like [built_value](https://github.com/google/built_value.dart/blob/master/built_value_generator/lib/built_value_generator.dart).

**Comment by techouse** [comment]
Hey 👋 

Thanx for the bug report. Should be fixed in #507

**Comment by iXamDev** [comment]
Hi, yes the issue is resolved.
Thanks a lot for your rapid assistance 🙌
